ROADMAP.md

```diff
@@ -0,0 +1,116 @@
+    short term, we'd like to define the Proxy-Wasm ABI in
+    [WIT](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md),
+    to understand:
+    -   How does the ABI surface map to WASI types and components?
+    -   Can WASI ABIs wholly replace Proxy-Wasm ABIs? Is performance on par?
+    -   Is there room for language SDKs, or is wit-bindgen the future?
+    -   How should we evolve Proxy-Wasm to become WASI-compatible? What are good
+        incremental steps?
+    -   See also: https://github.com/krinkinmu/wasi-http
+-   (Help wanted) Evaluate uses of foreign functions to identify feature gaps.
+    -   For example, Envoy
+        [registers foreign functions](https://github.com/search?q=repo%3Aenvoyproxy%2Fenvoy%20RegisterForeignFunction&type=code)
+        for signature checking, compression, filter state, route cache, and CEL
+        expressions.
+    -   Are there similar extensions in Nginx? Apache Traffic Server?
+    -   When/how do we promote these FFIs? Do they become negotiated features or
+        full members of the ABI?
+
+<a name="sdks"></a>
+
+## SDKs / language support
+
+-   (@leonm1, done!) Fork the abandoned Go SDK + support full Golang.
+-   (Google exploring) Build a Python SDK using a MicroPython port.
+-   (Help wanted) Stop using Emscripten in the C++ SDK. Instead use Clang /
+    wasi-sdk (proxy-wasm/proxy-wasm-cpp-sdk#167).
+-   (Help wanted) Build a Lua SDK using a Lua interpreter. Creates a safer
+    alternative to Envoy's
+    [Lua filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter),
+    and could benefit Nginx's Lua-based [OpenResty](https://openresty.org/)
+    ecosystem.
+
+<a name="host"></a>
+
+## Host features
+
+-   (@mpwarres) CppHost maintenance.
+    -   Update v8 and upstream some patches for v8 warming / extension.
+    -   Update the protobuf dependency.
+    -   Set up dependabot.
+-   (Help wanted) Prototype
+    [HyperLight](https://github.com/hyperlight-dev/hyperlight-wasm) as a
+    KVM-based sandboxing layer around wasm runtimes. The allure is getting an
+    inexpensive and transparent second layer of security at a thread boundary,
+    which makes it more feasible to run fully untrusted workloads with
+    Proxy-Wasm.
+-   (Help wanted) Performance benchmarks. One of Proxy-Wasm's strengths is its
+    ability to swap between multiple wasm runtimes. Help users make an informed
+    decision by benchmarking cold start and execution costs across runtimes.
+-   (Help wanted) Adopt CPU metering as a first-class feature. Leverage
+    instruction counting where available. For other engines (e.g. v8), use a
+    watchdog thread.
+-   (Help wanted) Support dynamic (per VM) limits for RAM and CPU.
+-   (Help wanted) Expand the use of SharedArrayBuffer to reduce memcpy into wasm
+    runtimes. This is promising for HTTP body chunks (see relevant
+    [WASI issue](https://github.com/WebAssembly/WASI/issues/594)) and
```

Comment:
This is unlikely to ever work, without host being specifically architected to read data from the network directly into Wasm's linear memory block.

AFAIK, the "multiple memories" proposal doesn't allow to attach / detach memories on-the-fly.